openpype/plugins/publish/integrate_new.py

```diff
@@ -923,15 +923,53 @@ def create_version_data(self, context, instance):
 
         # Include optional data if present in
         optionals = [
-            "frameStart", "frameEnd", "step", "handles",
-            "handleEnd", "handleStart", "sourceHashes"
+            "step", "handles", "sourceHashes"
         ]
         for key in optionals:
             if key in instance.data:
                 version_data[key] = instance.data[key]
+        version_data.update(self._get_optional_frame_data(instance))
 
         return version_data
 
+    def _get_optional_frame_data(self, instance):
+        anatomy = instance.context.data["anatomy"]
+        optional_keys = (
+            "frameStart", "frameEnd", "handleEnd", "handleStart"
+        )
+        optional_values = {}
+        for key in optional_keys:
+            value = instance.data.get(key)
+            if value is not None:
+                optional_values[key] = value
+
+        # Skip if frame start of end are not available
+        if (
+            "frameStart" not in optional_values
+            or "frameEnd" not in optional_values
+        ):
+            return optional_values
+
+        # Skip if both handles are not available
+        if (
+            "handleStart" not in optional_values
+            and "handleEnd" not in optional_values
+        ):
+            return optional_values
+
+        frame_info = UnifiedFrameInfo(
+            optional_values["frameStart"],
+            optional_values["frameEnd"],
+            optional_values.get("handleStart"),
+            optional_values.get("handleEnd"),
+            anatomy
+        )
+        optional_values["frameStart"] = frame_info.real_frame_start
+        optional_values["frameEnd"] = frame_info.real_frame_end
+        optional_values["handleStart"] = frame_info.handle_start
+        optional_values["handleEnd"] = frame_info.handle_end
```

Comment:
I think this function or just above these lines it's best to have a comment in the code as to why this is necessary. Something like:
```# We always store the frame start and frame end into the backend with handles excluded (extend) so we need to convert based on how the project is set.
```

Also, I'd change the logic so that only during Collect we retrieve the `UnifiedFrameInfo` and pass that along with the instance as `instance.data["frameInfo"]` or alike and then we don't need to duplicate the logic and build it again here. Or why would we want this code here too?

Reply:
Because these are data that were collected on instance and may not be the same as are on asset document.

Reply:
> Because these are data that were collected on instance and may not be the same as are on asset document.

I understand why we are doing the conversion at the end. I am just saying that we'd already construct the frame info during collect and use that one here too

Reply:
> I am just saying that we'd already construct the frame info during collect and use that one here too

Not exactly. Frame info on instance can be collected from workfile/scene and can be different than frame info from asset.

Reply:
Modified it and version document always use extending handles.